libs/blocks/merch/merch.js

```diff
@@ -116,17 +124,67 @@ export async function getPriceContext(el, params) {
   };
 }
 
+let loadEntitlementsPromise;
+
+export async function loadEntitlements() {
+  await loadIms();
+  if (!window.adobeIMS?.isSignedInUser?.()) return Promise.resolve([{}, {}]);
+  const [{ default: getUserEntitlements },
+    mappings] = await Promise.all([
+    import('../global-navigation/utilities/getUserEntitlements.js'),
+    fetch(entitlementsURL),
+  ]);
+  const entitlements = await getUserEntitlements();
+  const entitlementsMetadata = await mappings.json();
+  return [entitlements, entitlementsMetadata];
+}
+
+export async function handleEntitlements(cta, promise) {
+  try {
+    const [{ arrangement_codes: aCodes }, entitlementsMappings] = await promise;
+    if (aCodes !== undefined) {
+      await cta.onceSettled();
+      const { value: [{ productArrangementCode }] } = cta;
+      if (aCodes[productArrangementCode] === true) {
```

Comment:
right now you are comparing by arrangementCode, will it be the case also in future?
or we will switch to productFamily?

if we switch to product family you could already now get the family on an entitlement, without using the mapping. I think we need mapping only for the cta offer, but not for user entitlements.

```
const entitlements = await getUserEntitlements({ params: [{ name: 'include', value: 'OFFER.PRODUCT_ARRANGEMENT' }], format: 'raw' });

....

entitlements.map((offer) => offer.offer.product_arrangement.family)

```

Reply:
As soon as ODM-15772 is ready(pandora clientlibs must be updated as well), we can reduce it to a comparison of product families IMO.

But for now, I'll keep it as is.